PixelHunter1995/WalkingAreaLib/PolygonPartition.cs

```diff
@@ -70,5 +71,38 @@ public void Draw(GraphicsDeviceManager graphics, SpriteBatch spriteBatch, double
         {
             Polygons.ForEach(p => p.Draw(graphics, spriteBatch, scaling, sceneWidth));
         }
+
+        public bool Contains(Vector2 position)
+        {
+            return Polygons.Any(polygon => polygon.Contains(position));
+        }
+
+        /// <summary>
+        /// Get next position to walk towards.
+        /// </summary>
+        public Vector2 GetNextPosition(Vector2 clickPosition, Vector2 currentPosition)
```

Comment:
I feel like logically this method should belong in WalkingArea, though I don't know how many variables you would have to make public for that to happen...